enarx-keep-sev/src/vm.rs

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::x86_64::*;
+
+use std::error::Error;
+
+use kvm_bindings::kvm_userspace_memory_region;
+use kvm_ioctls::{Kvm, VmFd};
+use memmap::*;
+use x86_64::structures::paging::page_table::PageTableFlags;
+use x86_64::PhysAddr;
+
+pub struct VirtualMachine {
+    kvm: Kvm,
+    kvm_fd: VmFd,
+    address_space: Vec<kvm_userspace_memory_region>,
+    mmap_handles: Vec<MmapMut>,
+}
+
+impl VirtualMachine {
+    pub fn new(mem_size: usize) -> Result<Self, Box<dyn Error>> {
+        // Create a KVM context
+        let kvm = Kvm::new()?;
+        let kvm_fd = kvm.create_vm()?;
+
+        // Create the guest address space
+        let guest_addr_space = memmap::MmapOptions::new().len(mem_size).map_anon()?;
+        let region = kvm_userspace_memory_region {
+            slot: 0,
+            flags: 0,
+            guest_phys_addr: 0,
+            memory_size: mem_size as _,
+            userspace_addr: &*guest_addr_space as *const _ as *const u8 as u64,
```

Comment:
```suggestion
            userspace_addr: &*guest_addr_space as *const _ as u64,
```

Reply:
That was actually my initial stab at it, but I had offended the compiler:

```rust
error[E0606]: casting `*const [u8]` as `u64` is invalid
  --> src/vm.rs:37:29
   |
37 |             userspace_addr: &*guest_addr_space as *const _ as u64,
   |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: cast through a thin pointer first
```